libavpipe/src/avpipe_format.c

```diff
@@ -212,3 +212,53 @@ packet_clone(
     return 0;
 }
 
+/*
+ * Update an audio frame by skipping the necessary number of samples.
+ * Return the remaining samples in the frame (which may be 0 if the frame is to be skipped entirely)
+ * Many audio decoders use the concept of 'priming' and require skipping of a number of samples.
+ * Opus specifically sets codec_context->delay to the number of samples to be skipped.
+ * In most cases the audio frames to be skipped might have a negative PTS but that is not a rule/standard.
+ * Limitation: we expect we only skip samples in the first frame (log warning if not true)
+ *
+ * PENDING(SS) Currently unused but will be used in encode_frame()
+ */
+int
+audio_skip_samples(
+    AVCodecContext *codec_ctx,
+    AVFrame *frame) {
+
+    // Only skipping for Opus inputs currently
+    if (codec_ctx->codec_id != AV_CODEC_ID_OPUS) {
+        return frame->nb_samples;
+    }
+
+    int samples_to_skip = codec_ctx->delay; // Defined by the Opus format
+    if (samples_to_skip <= 0) {
+        return frame->nb_samples;
+    }
+
+    if (samples_to_skip > frame->nb_samples) {
+        // Unexpected - we need to skip more than one frame
+        elv_warn("Unexpected audio stream delay nb_frames=%d skip_now=%d", frame->nb_samples, samples_to_skip);
+        frame->nb_samples = 0;
+        return 0; // Entire frame needs to be skipped
+    }
+
+    int sample_size = av_get_bytes_per_sample(codec_ctx->sample_fmt);
+    if (sample_size <= 0) {
+        // Unknown sample format - this frame is not good
+        return eav_receive_frame;
```

Comment:
`eav_` errors are positive numbers, so we will need to either negate this or pass in an `int *` to hold the return value of the sample count. Otherwise this is indistinguishable from having 18 samples.

Reply:
Fixed!